src/black/__init__.py

```diff
@@ -2441,10 +2461,14 @@ def whitespace(leaf: Leaf, *, complex_subscript: bool) -> str:  # noqa: C901
 
             prevp_parent = prevp.parent
             assert prevp_parent is not None
-            if prevp.type == token.COLON and prevp_parent.type in {
-                syms.subscript,
-                syms.sliceop,
-            }:
+            if (
+                prevp.type == token.COLON
+                and prevp_parent.type
+                in {
+                    syms.subscript,
+                    syms.sliceop,
```

Comment:
This interacts a bit poorly with the magic trailing comma; we could have put this all in one line.

Reply:
I would like to address those things once we have decided how to best continue with this PR.